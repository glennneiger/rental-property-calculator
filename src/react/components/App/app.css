@import "../../../styles.css";

.app {
  font-family: var(--mainFontFamily);
}

button, input {
  font-family: var(--mainFontFamily);
}