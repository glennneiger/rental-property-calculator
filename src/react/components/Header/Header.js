import * as React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FaBars, FaWindowClose } from 'react-icons/fa';

const css = require('./header.css');
import HeaderAuthInfo from '../HeaderAuthInfo';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPath: this.props.location.pathname
    };
  }
  componentWillMount = () => {
    this.unlisten = this.props.history.listen(location => {
      this.setState({
        currentPath: location.pathname
      });
    });
  }
  componentWillUnmount = () => {
    this.unlisten();
  }
  getToggleSidebarIcon = () => {
    if (this.state.currentPath === '/') {
      return (this.props.sidebarVisible
        ? <FaWindowClose
          className={css.sidebarToggler}
          onClick={this.handleToggleSidebarClick}
        />
        : <FaBars
          className={css.sidebarToggler}
          onClick={this.handleToggleSidebarClick}
        />);
    }
    return null;
  }
  handleToggleSidebarClick = () => {
    this.props.sidebarVisible
      ? this.props.hideSidebar()
      : this.props.showSidebar();
  }
  render() {
    const title = (this.props.screenWidth > 420)
      ? 'Rental Property Calculator'
      : 'Calculator';

    return (
      <header className={css.header}>
        {this.getToggleSidebarIcon()}
        <Link to='/'>
          <span className={css.title}>
            {title}
          </span>
        </Link>
        <HeaderAuthInfo />
      </header>
    );
  }
}

// Header.propTypes = {
//   hideSidebar: PropTypes.func.isRequired,
//   history: PropTypes.object,
//   location: PropTypes.object.isRequired,
//   screenWidth: PropTypes.number.isRequired,
//   showSidebar: PropTypes.func.isRequired,
//   sidebarVisible: PropTypes.bool.isRequired
// };

export default Header;