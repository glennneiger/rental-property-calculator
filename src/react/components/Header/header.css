@import "../../../styles.css";

.header {
  background-color: var(--primaryColor);
  color: var(--textColorLight);
  height: var(--headerHeight);
  display: flex;
  align-items: center;
  justify-content: center;
}

.header {
  & a {
    text-decoration: none;
    color: var(--textColorLight);
    font-size: 1.8em;
    white-space: nowrap;
  }
}

.title {
  font-family: 'Dosis', sans-serif;
}

.sidebarToggler {
  position: absolute;
  cursor: pointer;
  left: 12px;
  min-width: 24px;
  min-height: 24px;
}

.authButtons {
  display: none;
}

.userInfo {
  display: none;
}

@media (--tablet-landscape-up) {
  .header {
    justify-content: space-between;
  }

  .userInfo {
    display: flex;
    align-items: center;

    & .userImage {
      font-size: 28px;
    }

    & p {
      margin-left: 8px;
      font-size: 18px;
    }
  }

  .header {
    & a {
      margin-left: calc(var(--calculatorContentPadding) + var(--calculatorInputSectionPadding));
    }
  }

  .authButtons {
    display: block;
    margin-right: 20px;
  }

  #loginButton {
    margin-right: 20px;
  }
}