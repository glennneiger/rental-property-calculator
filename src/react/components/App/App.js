import * as React from 'react';
import { Route } from 'react-router-dom';
import * as jwtDecode from 'jwt-decode';
import * as Modal from 'react-modal';

import Header from '../Header';
import ModalRoot from '../Modals/ModalRoot';
import CalculatorPage from '../CalculatorPage';
import LoginPage from '../Authentication/LoginPage';
import RegisterPage from '../Authentication/RegisterPage';
import { setAuthToken } from '../../../utils/authUtils';
import store from '../../../store';
const css = require('./app.css');
import {
  logoutUser,
  setCurrentUser
} from '../../../actions/auth';
import {
  MILLISECONDS_PER_SECOND
} from '../../../constants';

if (localStorage.jwtToken) {
  setAuthToken(localStorage.jwtToken);
  const decodedToken = jwtDecode(localStorage.jwtToken);
  store.dispatch(setCurrentUser(decodedToken));
  const currentTime = Date.now / MILLISECONDS_PER_SECOND;
  if (decodedToken.exp < currentTime) {
    store.dispatch(logoutUser());
    window.location.href = '/login';
  }
}

class App extends React.Component {
  componentDidMount() {
    Modal.setAppElement('#app');
  }
  render() {
    return (
      <div className={css.app} id='app'>
        <ModalRoot />
        <Header />
        <Route exact path='/' component={CalculatorPage} />
        <Route exact path='/login' component={LoginPage} />
        <Route exact path='/register' component={RegisterPage} />
      </div>
    );
  }
}

export default App;