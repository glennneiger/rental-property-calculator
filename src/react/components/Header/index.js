import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from './Header';
import {
  hideSidebar,
  showSidebar
} from '../../../actions/sidebarVisible';


const mapStateToProps = state => ({
  screenWidth: state.ui.screenWidth,
  sidebarVisible: state.sidebarVisible
});

const mapDispatchToProps = dispatch => ({
  hideSidebar: () => {
    dispatch(hideSidebar());
  },
  showSidebar: () => {
    dispatch(showSidebar());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Header));