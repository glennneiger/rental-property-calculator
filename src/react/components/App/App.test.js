import * as React from 'react';
import { expect as chaiExpect } from 'chai';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';
import Header from '../Header';

describe('<App />', () => {
  let componentDidMountMock;

  beforeAll(() => {
    componentDidMountMock = jest.spyOn(
      App.prototype,
      'componentDidMount'
    ).mockImplementation();
  });

  afterAll(() => {
    componentDidMountMock.mockRestore();
  });

  it('has proper className for styling', () => {
    const wrapper = shallow(<App />);

    chaiExpect(wrapper).to.have.className('app');
  });

  it('has proper id', () => {
    const wrapper = shallow(<App />);

    chaiExpect(wrapper).to.have.id('app');
  });

  it('should have three routes', () => {
    const wrapper = shallow(<App />);

    chaiExpect(wrapper.find('Route')).to.have.length(3);
  });

  it('renders a header component', () => {
    const wrapper = shallow(<App />);

    chaiExpect(wrapper.find(Header)).to.have.length(1);
  });

  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<App />);

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });
});