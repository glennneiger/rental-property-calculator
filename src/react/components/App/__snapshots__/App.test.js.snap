// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
<div
  className="app"
  id="app"
>
  <Connect(ModalRoot) />
  <Connect(withRouter(Header)) />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/register"
  />
</div>
`;
